server/controllers/request.js

```diff
@@ -75,18 +104,36 @@ exports.updateRequest = asyncHandler(async (req, res, next) => {
   const isOwnedByUser = request.owner == userId || request.sitter == userId;
 
   if (isOwnedByUser) {
-    Object.entries(input).forEach(([key, value]) => {
-      if (value) {
-        if (key === "startDate") {
-          request.startDate = new Date(value);
+    await Promise.all(
+      Object.entries(input).map(async ([key, value]) => {
+        if (value) {
+          if (key === "status" && value === "PAID") return;
+          else if (key === "status" && value === "ACCEPTED") {
+            const charge = await stripe.charges.capture(request.chargeId);
+            if (charge.status === "failed") {
+              res.status(400);
+            } else {
+              request.status = "PAID";
+            }
+          } else if (key === "status" && value === "DECLINED") {
+            if (!request.chargeId) {
+              res.status(400);
+            }
+            await stripe.refunds.create({
+              charge: request.chargeId,
+            });
+            request.status = "DECLINED";
+            request.chargeId = null;
+          } else if (key === "startDate") {
+            request.startDate = new Date(value);
+          } else if (key === "endDate") {
+            request.endDate = new Date(value);
+          } else {
+            request[key] = value;
+          }
```

Comment:
It would be a cleaner code to first map all the Promises and assign them to a variable, then pass the variable to `Promise.all`.